explainer.md

```diff
@@ -219,78 +226,159 @@ user.
+When the element is painted, stashed draw commands are used.
+
+##### Committing the lock
 
-* The updates are processed as long as they do not cause undue delay to the
-  rest of the update phases
+Unless the lock's durating is extended (see below), the lock is committed and
+the following operations take place:
+
+* The updates to layout and paint are processed as long as they do not cause
+  undue delay to the rest of the update phases
 * If an undue delay is likely to be caused, the work already completed is
   processed and the update phase yields to other update phases for unlocked
   content.
 * Future update phases for the locked element are gated on the fact that the
   previous update phases for this locked element have completed.
-* When the element is painted, stashed draw commands are used. In other words,
-  new draw commands could be generated and stored but not presented to the
-  screen.
-
-##### DisplayLock.commit() [*bikeshed: release()*]
-
-Commit performs the following steps:
-
-* It returns a promise, which resolves when the element enters the unlock
-  pending phase.
-* It finishes all update phases for the locked element, as described above.
-* It enters the unlock pending phase and return the promise. Note that at this
-  point, the element is still locked for visual updates. This allows the
-  callback in the promise resolution to re-lock the lock without changing the
-  visual representation of the element.
-* After the script phase is done and the lock has not been re-acquired, the
-  element is unlocked and enters the element is unlocked state.
+* As before, when the element is painted, stashed draw commands are used. Note
+  that the user-agent is free to paint the element with the updated visual
+  representation, as long as that representation is not presented to screen.
+* When all of the above phases finish, the promise returned by
+  `acquireDisplayLock()` is resolved, giving the script opportunity to
+  re-acquire the lock before the contents are displayed.
+  * If the promise resolution callback does not re-acquire the lock, the
+    contents are presented to screen.
+  * If the pormise resolution callback does re-acquire the lock, then the
+    process starts again in a locked state and the visual contents remain
+    unchanged (i.e. the contents displayed are the same contents as before the
+    intial lock was acquired). (Note: see active vs passive locks discussions
+    below)
+
+##### Extending the lifetime of the lock
+
+It is possible that the passed callback, which is run to completion, runs for a
+long enough time to actually cause jank: it may prevent other script from
+running. A way to alleviate this is to schedule a continuation of the work using
+a DisplayLockContext, which is given to the callback.
+
+##### DisplayLockContext.schedule()
+
+The `acquireDisplayLock()` callback may call `context.schedule()` passing it
+another callback which will schedule to run on the next frame. This allows for
```

Comment:
"which will run as a separate event loop task in the future"

We don't map these directly to rendering.